README.md

```diff
@@ -1,189 +1,337 @@
+you want to perform 7 batches of a 256 x 256 transform.
 
-For example, if there is a single input `r` and output `s`:
+Since the transform is a 2D one, the user creates a interface with `itype` of
+dimension N+1: in this case, an `int3d`. The last dimension is used to store the
+number of batches.
 
+```lua
+local fft2d_batch_complex64_complex64 = fft.generate_fft_interface(int3d, complex64, complex64)
 ```
-fft1d.execute_plan(r, s, p)
+
+The input and output regions should be 256 x 256 x 7 arrays: i.e., the last
+dimension is the number of batches. The plan region remains the same as before:
+
+```lua
+var r = region(ispace(int3d, {256, 256, 7}), complex64)
+var s = region(ispace(int3d, {256, 256, 7}), complex64)
+var p = region(ispace(int1d, 1), fft3d_batch_real.plan)
 ```
 
-Or if there are partitions `r_part` and `s_part` that are distributed
-around the machine, one might do:
+The key difference is that we call `make_plan_batch` instead of `make_plan`
 
+```lua
+fft2d_batch_complex64_complex64.make_plan_batch(r, s, p)
 ```
-__demand(__index_launch)
-for i in r_part.colors do
-  fft1d.execute_plan_task(r_part[i], s_part[i], p)
-end
+
+Then, we execute and destroy as in the regular case.
+
+```lua
+fft2d_batch_complex64_complex64.execute_plan_task(r, s, p)
+fft2d_batch_complex64_complex64.destroy_plan(p)
 ```
 
-Note that `execute_plan` is a `__demand(__inline)` task (as described
-above). The task `execute_plan_task` is simply a wrapper around
-`execute_plan` for convenience, to avoid needing to define this
-explicitly.
+As you can see, the main points of differentiation from the regular transform
+API is that we input a region of dimension `n+1`, where the final dimension is
+the number of batches, and use `make_plan_batch` instead of `make_plan`
 
-**Important:** because `execute_plan` is a `__demand(__inline)` task,
-it will never execute on the GPU (unless the parent task is running on
-the GPU). Therefore, in most cases it is necessary to use
-`execute_plan_task` if one wants to use the GPU.
+Please also refer to the `test_2d_complex64_to_complex64_batch_transform` and
+`test_2d_double_to_complex64_batch_transform` examples in `fft_test.rg` for
+reference.
 
-**Important:** while `execute_plan_task` may be executed on the GPU,
-the contents of the `p` region must still be available on the CPU,
-because the plans must be used by the host-side code to launch the FFT
-kernels. Therefore, when `execute_plan_task` is mapped onto the GPU it
-is very important to map the `p` region into zero-copy memory.
+Batched transforms are supported on both CPU and GPUs, for 1, 2 and 3
+dimensions. Both single-precision and double-precision real-to-complex and
+complex-to-complex transforms are supported.
 
-### Finalization
+#### 6. Distributed Mode
 
-When a plan is no longer needed it can be distroyed.
+The API also supports a distributed mode, where every machine in a distributed
+job executes an independent FFT of the same size.
```

Comment:
I'm not sure this is actually true? If you partitioned the input/output regions non-uniformly I think everything would work fine.

Reply:
I think I took this from the 'implementation notes' section of the original readme. I removed the 'of the same size' portion in any case, so hopefully this is accurate now.